src/main/java/com/example/animalshelter/model/Animal.java

```diff
@@ -1,25 +1,35 @@
 package com.example.animalshelter.model;
 
-import lombok.Builder;
-import lombok.Data;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
 
-import javax.validation.constraints.NotNull;
-import javax.validation.constraints.Positive;
+import javax.persistence.*;
 
 /**
  * Keeps basic information about an animal
  */
-@Data
-@Builder
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
 public class Animal {
 
-    @NotNull
-    @Positive
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
     private Long id;
-    
+
     private String name;
+
     private int age;
+
+    @Enumerated(EnumType.STRING)
     private AnimalType type;
+
+    @Enumerated(EnumType.STRING)
     private AnimalGender gender;
+
+    @Enumerated(EnumType.STRING)
     private AnimalHealthStatus healthStatus;
+
```

Comment:
How the equality of two Animals are now checked?
Add test for saving 'same' Animal twice.

Reply:
I have tested this and it looks that if we provide two `Animal` object with same id the second animal override the first one, however I think that this shouldn't happen because when we save animal we do not provide id - it is generated for us.

And in the second scenario when we provide animal without id it is saved as a new record with different id.

I was looking at this article: https://thorben-janssen.com/ultimate-guide-to-implementing-equals-and-hashcode-with-hibernate/  and it seems that using id in equals and has code is not preferred.

> As I teased earlier, generated primary keys create a problem for the implementation of your equals() and hashCode() methods. That’s because the primary key value gets set when the entity gets persisted. So, your entity object can exist with and without a primary key value.

Here is an article about equals and hashcode https://vladmihalcea.com/how-to-implement-equals-and-hashcode-using-the-jpa-entity-identifier/ and the proposed solution for implementation is set hascode as a constant value.

> The entity identifier can be used for equals and hashCode, but only if the hashCode returns the same value all the time. This might sound like a terrible thing to do since it defeats the purpose of using multiple buckets in a HashSet or HashMap.
> 
> However, for performance reasons, you should always limit the number of entities that are stored in a collection. You should never fetch thousands of entities in a @OneToMany Set because the performance penalty on the database side is multiple orders of magnitude higher than using a single hashed bucket.

We discussed with Patrycja that we might use NaturalId(business key) in the future. It might be a unique leash id for a animal. We may base equals and hashcode implementation on that value.

Still looking into this.

Reply:
I have added implementation for `equals()` and `hashcode()` methods in the `Animal` class, and a few tests. @BulaMar  Please take a look when you find some time, I stay with option that the animal Id should not be part of equals and hashcode methods (as it might not be present). There were some other changes to. I have added a new tests for integration with a database (I have tested the possibility of save same animal twice in that way). 

Please let me know what do you think about this. Thank you!